vcs2l/commands/import_.py

```diff
@@ -86,23 +87,92 @@ def file_or_url_type(value):
     try:
-        root = yaml.safe_load(yaml_file)
+        return yaml.safe_load(yaml_file)
     except yaml.YAMLError as e:
-        raise RuntimeError('Input data is not valid yaml format: %s' % e)
+        raise RuntimeError('Input data is not valid yaml format: %s' % e) from e
 
+
+def get_repositories_from_root(root):
+    """Extract repositories from the parsed YAML root object."""
     try:
         repositories = root['repositories']
         return get_repos_in_vcs2l_format(repositories)
     except KeyError as e:
-        raise RuntimeError('Input data is not valid format: %s' % e)
+        raise RuntimeError('Input data is not valid format: %s' % e) from e
     except TypeError as e:
         # try rosinstall file format
         try:
             return get_repos_in_rosinstall_format(root)
         except Exception:
-            raise RuntimeError('Input data is not valid format: %s' % e)
+            raise RuntimeError('Input data is not valid format: %s' % e) from e
+
+
+def get_repositories(yaml_file, visited_files=None):
+    """Recursively get repositories from a YAML file, handling inheritance."""
+    if visited_files is None:
+        visited_files = set()
+
+    # Get absolute path to handle relative paths consistently
+    current_file_path = os.path.abspath(yaml_file.name)
+
+    if current_file_path in visited_files:
+        raise CircularImportError(f'Circular import detected: {current_file_path}')
+
+    visited_files.add(current_file_path)
+
+    try:
+        root = load_yaml_file(yaml_file)
+
+        combined_repos = {}
+
+        if 'extends' in root:
+            parent_files = root['extends']
+            # Convert single file to list for consistent processing
+            if isinstance(parent_files, str):
+                parent_files = [parent_files]
+
+            # Check for duplicate entries in extends
+            if len(parent_files) != len(set(parent_files)):
+                raise RuntimeError(
+                    f'Duplicate entries found in extends in file: {current_file_path}'
+                )
+
+            for parent_file in parent_files:
+                # If absolute path is not valid, try relative to current file
+                if not os.path.isabs(parent_file):
+                    current_dir = os.path.dirname(current_file_path)
+                    parent_file = os.path.join(current_dir, parent_file)
+
+                if not os.path.exists(parent_file):
+                    raise RuntimeError(f'Parent file not found: {parent_file}')
+
+                try:
+                    # Recursively get repositories from parent file
+                    with open(parent_file, 'r', encoding='utf-8') as parent_f:
+                        parent_repos = get_repositories(parent_f, visited_files.copy())
+                        combined_repos.update(parent_repos)
+                except Exception as e:
+                    if isinstance(e, CircularImportError):
+                        raise
```

Comment:
Minor suggestion: I think you can do this and not have the `CircularImportError` be caught below:

```suggestion
                except CircularImportError:
                    raise
                except Exception as e:
```